### ISSUE TITLE
Fallback to default cache time for Cache::put when minutes parameter is not provided

### ISSUE THREAD

**Issue by Codeklopper** [laravel/internals#163]
### TL;DR

Laravel version: 5.3
Php version: 5.3
Class: \Illuminate\Cache\Repository
Used cache driver: File and Redis

Command: `Cache::put('a', 'b')`
Expected result: Item stored in the cache, for default time. Or an exception when item could not be stored
Actual result: Item not stored in cache
### Explanation

When I used `Cache::put('a', 'b')` without providing the `$minutes` parameter, the item was not stored in the cache. Because the $minutes parameter is optional, I expected it would fall back to the default cache time (which is set to 60 minutes) or the time I would set with `setDefaultCacheTime`, like ofssetSet (but with an optional `$minutes` parameter): 

``` php
public function offsetSet($key, $value)
    {
        $this->put($key, $value, $this->default);
    }
```

I saw the reason why the `$minutes` parameter is optional, is because of the option to pass an array and handle the put call as a putMany call. However, calling putMany without `$minutes` is not possible, because the parameter is not optional.

So, should the put method be changed to store the item for the default cache time or return a result like the add method? For example:

``` php
public function put($key, $value, $minutes = null)
    {
        if (is_array($key)) {
            return $this->putMany($key, $value);  // $value used as minutes, cannot be null
        }

        // shorthand solution, other solutions if $minutes are not provided could be to throw an exception  or return a boolean
        $minutes = $this->getMinutes(is_null($minutes) ? $this->default : $minutes);

        if (! is_null($minutes)) {
            $this->store->put($this->itemKey($key), $value, $minutes);

            $this->fireCacheEvent('write', [$key, $value, $minutes]);
        }
    }
```
